### ISSUE TITLE
Non-strict comparison of md5() hash 

### ISSUE THREAD

**Issue by elsand** [bambora/checkout-woocommerce#1]
`successful_request()` uses `md5()` in an insecure way. Due to PHPs type juggling, comparing hashes should always be done strictly (with ===), or better yet, with [`hash_equals()`](http://php.net/hash_equals), or else non-equal strings can be parsed as numeric, converted to 0, and thus be viewed as equal, enabling an attacker to supply invalid hashes to break the verification. Consider:

```
$ php -r "echo md5('QNKCDZO') . PHP_EOL;"
0e830400451993494058024219903391
$ php -r "echo md5('aabC9RqS') . PHP_EOL;"
0e041022518165728065344349536299
$ php -r "var_dump(md5('aabC9RqS') == md5('QNKCDZO')) . PHP_EOL;"
bool(true)´
$ php -r "var_dump(md5('aabC9RqS') === md5('QNKCDZO')) . PHP_EOL;"
bool(false)
```
See http://www.cryptologie.net/article/268/how-to-compare-password-hashes-in-php for more information.

**Comment by AllanWLie** [comment]
Thank you for reporting this issue. We will look into it and consider it for the next release. In the meantime feel free to make a pull request and implement the issues.

**Comment by sorenla** [comment]
Fixed in version 1.2.8
